In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px
import plotly.graph_objects as go
import ipywidgets as widgets

In [3]:
df = pd.read_csv ('ForeignGifts_edu.csv')

# UVA Findings

In [4]:
uva_df = df[df['Institution Name'] == 'University of Virginia']

In [5]:
#Count of foreign gifts to UVA
print(uva_df.shape[0])

#Total amount of foreign gifts to UVA
print(uva_df['Foreign Gift Amount'].sum())

#Average amount of foreign gifts to UVA
print(uva_df['Foreign Gift Amount'].mean())

98
22189238
226420.79591836734


The output of the code above shows that UVA has recieved 98 total foreign gifts totaling to $22,189,238. The average amount per foreign gift for UVA was about $226,421

The above output shows that the top gift type for UVA in terms of amount of gifting was contracts which provided almost double the dollar amount of gifting compated to monetary gifts.

In [6]:
#Histogram of UVA foreign gift amount
gift_amount_hist = px.histogram(uva_df, x='Foreign Gift Amount', title='Distribution of Amount per Foreign Gift at UVA', labels={'Foreign Gift Amount': 'Foreign Gift Amount ($)'})
gift_amount_hist.update_layout(yaxis_title='Count of Foreign Gifts')

gift_amount_hist.show()

When looking at the distribution of foreign gift amounts to UVA, the graph above shows that the majority of foreign gifts that UVA recieves are less than $200,000.

In [ ]:
#Histogram of country of giftor to UVA
uva_country_hist = px.histogram(uva_df, x='Country of Giftor', 
                                title='Count of Foreign Gifts by Country',
                                labels={'Country of Giftor': 'Country'})
uva_country_hist.update_layout(yaxis_title='Count of Foreign Gifts')

#Bar of gift amount to UVA by country
uva_gift_amount_country = uva_df.groupby('Country of Giftor', as_index=False).sum("Foreign Gift Amount")
uva_gift_amount_bar = px.bar(uva_gift_amount_country, x='Country of Giftor', 
                             y='Foreign Gift Amount', 
                             title='Total Foreign Gift Amount by Country', 
                             labels={'Foreign Gift Amount': 'Total Gift Amount ($)', 'Country of Giftor': 'Country'})

uva_country_hist.show()
uva_gift_amount_bar.show()

Breaking this down by where these gifts are coming from, the graph titled 'Count of Foreign Gifts by Country' shows that UVA recieves the most gifts by far from England. But it can be seen in the second graph titled 'Total Foreign Gift Amount by Country', that Bangledesh and Sweeden have given UVA the most foreign gifting in dollar amount. These both have a total gifting amount that is at least double that of England.

In [8]:
#Breakdown of number of gifts by gift type
pd.DataFrame(uva_df['Gift Type'].value_counts().reset_index(name = 'Count of Foreign Gifts'))

,Gift Type,Count of Foreign Gifts
0,Contract,64
1,Monetary Gift,34


I next wanted to see what type of foreign gifts UVA was recieving at the highest volume. The table above shows that the largest number of foriegn gifts to UVA are contracts and the only other type of gift recieved is monetary gifts. 

In [9]:
uva_df.groupby("Gift Type").sum("Foreign Gift Amount").sort_values(by="Foreign Gift Amount", ascending=False)[['Foreign Gift Amount']]

,Foreign Gift Amount
Gift Type,
Contract,14807636
Monetary Gift,7381602


When looking at the gift amounts, UVA recieved a higher dollar amount in foreign gifting through contracts at $14,807,636 than through monetary gifting which only earned them $7,381,602.

# Comparison to Other Top Public Universities

In [10]:
#Creating data set for UVA and peer institutions
top_public_universities = ['University of Virginia', 'University of Michigan - Ann Arbor', 'University of North Carolina - Chapel Hill', 'University of California, Berkeley', 'University of California, Los Angeles', 'University of California, San Diego', 'University of Florida', 'University of Texas at Austin']
uva_and_peers_df = df[df['Institution Name'].isin(top_public_universities)]

To be able to put UVA's performance in foreign gifting into context I wanted to compare it to the performance of other top public universities. I chose 7 other universities to compare UVA to: 
- University of Michigan - Ann Arbor
- University of North Carolina - Chapel Hill
- University of California, Berkeley
- University of California, Los Angeles
- University of California, San Diego
- University of Florida, University of Texas at Austin.

In [11]:
#Number of foreign gifts by institution
uva_and_peers_df.groupby("Institution Name").sum("Foreign Gift Amount").sort_values(by="Foreign Gift Amount", ascending=False)[['Foreign Gift Amount']]

,Foreign Gift Amount
Institution Name,
"University of California, Berkeley",294479904
University of Michigan - Ann Arbor,287336783
"University of California, Los Angeles",242146363
"University of California, San Diego",131941755
University of Texas at Austin,85858408
University of North Carolina - Chapel Hill,41292544
University of Florida,24759139
University of Virginia,22189238


First comparing UVA's $22,189,238 total foreign gifting to other the other top public universities I selected, it ranks last. The top of these public universties in terms of total foreign gift amount is Berkely at $2,94,479,904 which is over 13 times that of UVA. University of Michigan - Ann Arbor comes in second at $287,336,783 which is still 12 times that of UVA.

In [12]:
pd.DataFrame(uva_and_peers_df['Institution Name'].value_counts().reset_index(name = 'Count of Foreign Gifts'))

,Institution Name,Count of Foreign Gifts
0,"University of California, Los Angeles",3916
1,University of Michigan - Ann Arbor,715
2,"University of California, Berkeley",394
3,University of Texas at Austin,377
4,University of North Carolina - Chapel Hill,286
5,"University of California, San Diego",203
6,University of Virginia,98
7,University of Florida,33


There is a similar story when looking at the total number of foreign gifts recieved by each institution where UVA is second-to-last followed only by University of Florida. University of California, Los Angeles recieved the largest number of foreign gifts at 3,916 and University of Michigan - Ann Arbor	came in second at 715.

In [ ]:
uvapeers_grouped = uva_and_peers_df.groupby('Country of ', as_index=False).sum("Foreign Gift Amount")
uvapeers_bar = px.bar(uvapeers_grouped, x='Institution Name', 
                        y='Foreign Gift Amount', 
                        title='Total Foreign Gift Amount by Institution', 
                        labels={'Foreign Gift Amount': 'Total Gift Amount ($)', 'Institution Name': 'Academic Institution'})


uvapeers_bar.show()

In [27]:
uvapeers_grouped  = uva_and_peers_df.groupby('Country of Giftor', as_index=False).sum("Foreign Gift Amount").sort_values(by='Foreign Gift Amount', ascending=False).head(20)
uvapeers_grouped_bar = px.bar(uvapeers_grouped, x='Country of Giftor', 
                             y='Foreign Gift Amount', 
                             title='Total Foreign Gift Amount by Country', 
                             labels={'Foreign Gift Amount': 'Total Gift Amount ($)', 'Country of Giftor': 'Country'})

uvapeers_grouped_bar.show()

I also wanted to see what countries give the most to the top public universities and the graph above shows that England provided the highest dollar amount of gifting to these universities.

# Recommendation

From the anlaysis of the data above it can be seen that UVA is falling below it's peers in terms of foreign giving and has room for improvemnet to increase the amount it is recieving in gifting. I noticed in the data that while England is the country that gives UVA the most individual gifts, it is not contributing the most to the total amount of foreign gifting to the institution. I looked further into the countries that are providing the highest total gift amount and what type of gifting they are recieving in the code below. Sweden and Bangledesh appear to give the most in the form of contracts to UVA, as well as some in monetary gifts. Because these contracts appear to provide a higher amount of total gifting, I also looked at a higher level accross all academic instiutions to see if contracts make up a higher proportion of total gifting overall as well. In the second table below, it shows that contracts make up over double the total foreign gifting than monetary contracts. This might indicate that it could be worth exploring further contract opporunities within England, which UVA already has a strong monetary gifting relationship with, and somewhat of contract giving history as well. It is also the case that England is the overall top foreign giftor to the top public institutions I looked at. This could allow them to build on these existing relationships and networks in order to increase the total gifting amount they receive.

In [14]:
uva_df.groupby(['Country of Giftor','Gift Type']).sum("Foreign Gift Amount").sort_values(by='Foreign Gift Amount', ascending=False)

,,ID,OPEID,Foreign Gift Received Date,Foreign Gift Amount
Country of Giftor,Gift Type,,,,
SWEDEN,Contract,262920,3745000,428079,4007552
BANGLADESH,Contract,157699,2247000,256206,3642743
TANZANIA,Contract,78918,1123500,129117,2776447
SWITZERLAND,Monetary Gift,52658,749000,87207,2000000
GUERNSEY,Monetary Gift,78951,1123500,130155,1500000
ENGLAND,Monetary Gift,446763,6366500,723373,1447508
FINLAND,Contract,184144,2621500,300867,1008384
BERMUDA,Monetary Gift,26279,374500,42502,900000
KOREA,Contract,157595,2247000,254053,862787


In [15]:
df.groupby("Gift Type").sum("Foreign Gift Amount").sort_values(by="Foreign Gift Amount", ascending=False)[['Foreign Gift Amount']]

,Foreign Gift Amount
Gift Type,
Contract,10809974431
Monetary Gift,5775853163
Real Estate,14687920
